script/app.js

```diff
@@ -139,3 +147,77 @@ const completeTodo = (taskId) => {
     todos = updatedTodos;
     renderTodos();
 };
+
+const editTodo = (taskId) => {
+    const task = todos.find((task) => task.id === taskId);
+    editingId = task.id;
+
+    if (!task) {
+        alert("Task not found");
+        return;
+    }
+
+    const taskElement = document.getElementById(`task-${task.id}`);
+    const titleElement = taskElement.querySelector("h1");
+    const createdAtElement = taskElement.querySelector(".createdAt");
+    const inputElement = document.createElement("input");
+    inputElement.type = "text";
+    inputElement.value = task.title;
+    if (titleElement) {
+        titleElement.parentNode.replaceChild(inputElement, titleElement);
+    }
+    if (createdAtElement) {
+        createdAtElement.style.display = "none";
+    }
+    const editButton = taskElement.querySelector(".task-card__icon--edit");
+    editButton.innerText = "Save";
+
+    const saveHandler = () => {
+        const updatedTitle = sanitizeInput(inputElement.value.trim());
+
+        if (!updatedTitle) {
+            alert("Task title cannot be empty");
+            return;
+        }
+
+        const updatedTodos = [...todos];
+        const taskToUpdate = updatedTodos.find((task) => task.id === taskId);
+        taskToUpdate.title = updatedTitle;
+        todos = updatedTodos;
+        renderTodos();
+        editButton.innerText = "Edit";
+        editButton.removeEventListener("click", saveHandler);
+        if (editingId === task.id) {
+            editingId = -1;
+            renderTodos();
+        }
+    };
+
+    const completeHandler = () => {
+        saveHandler();
+        completeTodo(task.id);
+    };
+
+    editButton.addEventListener("click", saveHandler);
+
+    const completeButton = taskElement.querySelector(
+        ".task-card__icon--complete"
+    );
+    completeButton.addEventListener("click", completeHandler);
+
+    const deleteHandler = () => {
```

Comment:
why is it needed? we are already deleting in deleteTodo function

Reply:
When we are in edit mode, the delete button should work as a cancel button. That's why we need this delete handle (on the basis of the icon). I may name it a cancel handler. Should I do that or keep it?